examples/ime_input.rs

```diff
@@ -0,0 +1,86 @@
+use log::LevelFilter;
+use simple_logger::SimpleLogger;
+use winit::{
+    event::{ElementState, Event, VirtualKeyCode, WindowEvent, IME},
+    event_loop::{ControlFlow, EventLoop},
+    window::WindowBuilder,
+};
+
+fn main() {
+    SimpleLogger::new()
+        .with_level(LevelFilter::Debug)
+        .init()
+        .unwrap();
+
+    println!("Press F2 to toggle IME. See the documentation of `set_ime_allowed` for more info");
+
+    let event_loop = EventLoop::new();
+
+    let window = WindowBuilder::new()
+        .with_inner_size(winit::dpi::LogicalSize::new(128.0, 128.0))
+        .build(&event_loop)
+        .unwrap();
+
+    let mut ime_allowed = true;
+    window.set_ime_allowed(ime_allowed);
+
+    let mut may_show_ime = false;
+    let mut window_pos = winit::dpi::PhysicalPosition::new(10, 10);
+
+    event_loop.run(move |event, _, control_flow| {
+        *control_flow = ControlFlow::Wait;
+        match event {
+            Event::WindowEvent {
+                event: WindowEvent::CloseRequested,
+                ..
+            } => *control_flow = ControlFlow::Exit,
+            Event::WindowEvent {
+                event: WindowEvent::Moved(new_pos),
+                ..
+            } => {
+                window_pos = new_pos;
+                if may_show_ime {
+                    println!("window moved {:?}", window_pos);
+                    window.set_ime_position(window_pos);
+                }
+            }
```

Comment:
It seems cumbersome that the user has to set the IME position every time the window moves - can we change `set_ime_position` so that this is not required?

Reply:
I used this function incorrectly. The function sets the IME position relative to the window, so this is not needed anyways. (Althought there's a bug that the candidate selection box appears at the wrong place then deleting the preedit text and typeing again makes it appear at the correct place. I should fix that later)